test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -720,11 +776,94 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
 
             // Assert
             // OrderBy is used because the order of the results may very depending on the platform / client.
-            Assert.Equal(
-                expectedKeyValuePairs?.OrderBy(item => item.Key.Group, StringComparer.Ordinal)
-                .ThenBy(item => item.Key.Name, StringComparer.Ordinal),
-                context.DisplayMetadata.EnumGroupedDisplayNamesAndValues?.OrderBy(item => item.Key.Group, StringComparer.Ordinal)
-                .ThenBy(item => item.Key.Name, StringComparer.Ordinal));
+            var expectedOrdered = expectedKeyValuePairs?
+                .OrderBy(item => item.Key.Group, StringComparer.Ordinal)
+                .ThenBy(item => item.Key.Name, StringComparer.Ordinal);
+            var actualOrdered = context.DisplayMetadata.EnumGroupedDisplayNamesAndValues?
+                .OrderBy(item => item.Key.Group, StringComparer.Ordinal)
+                .ThenBy(item => item.Key.Name, StringComparer.Ordinal);
```

Comment:
Put another way, fix the data set if `expectedKeyValuePairs` order is wrong. It'll always be wrong and the comment just above is also wrong.